### PR TITLE
RFC: Co-locate All Paketo RFCs

### PR DESCRIPTION
**Pull request by ForestEckhardt**
[Readable](https://github.com/ForestEckhardt/rfcs-1/blob/co-locate-all-rfcs/text/0027-co-locate-all-rfcs.md)

### CODE REVIEW

**Comment by fg-j** [comment]
@ForestEckhardt heads up there are 2 RFCs on this pull request. was that intentional?

File: text/0027-co-locate-all-rfcs.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+be centralized into the top level RFCS repository.
+
+## Motivation
+
+As it stands currently there are RFCs scattered across various repositories
+throughout the Paketo organization. These RFCs pertain typically to the
+implementation and functionality of either entire language families or even on
+down to implementation buildpack specific behavior. While it is good to have a
+record of technical decisions made for the project and its components, it is
+sometimes hard to search for relative RFCs for a given language family or
```
Comment (fg-j) [inline text/0027-co-locate-all-rfcs.md:15]:
```suggestion
sometimes hard to search for relevant RFCs for a given language family or
```

File: text/0027-co-locate-all-rfcs.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+feature. By co-locating all of the RFCs for the project into one repository we
+will collect all of the decision making conversation into one place making it
+easier to access.
+
+## Implementation
+
+Each project team will be given a directory inside of the `rfcs` repository,
+which would be named after the project team. A series of directories with the
```
Comment (fg-j) [inline text/0027-co-locate-all-rfcs.md:23]: Does "project team" here mean the teams within the Paketo Buildpacks Github org
(https://github.com/orgs/paketo-buildpacks/teams)?

File: text/0027-co-locate-all-rfcs.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+```
+rfcs
+├── assets
+│   └── ...
+├── LICENSE
+├── NOTICE
+├── README.md
+└── text
+    ├── 0000-template.md
+    ├── ...
```
Comment (fg-j) [inline text/0027-co-locate-all-rfcs.md:35]: I think this is sort of implied, but is it your intent that there are still RFCs outside of project subdirectories? These would be RFCs that need to be voted on by the whole project and not just a subteam, I'm guessing.

**Comment by fg-j** [comment]
One more thing I like about this is that an RFC can be "escalated" to a project-wide RFC or "demoted" into a language-family RFC just by moving it in the directory structure. Makes it easy to re-evaluate the scope of an RFC while it's under discussion.

File: text/0027-co-locate-all-rfcs.md (hunk 3 above)
Comment (ForestEckhardt) [reply text/0027-co-locate-all-rfcs.md:35]: The RFCs that are outside of the project subdirectories would be Paketo level RFCs that need to be approved by the steering committee.

File: text/0027-co-locate-all-rfcs.md (hunk 2 above)
Comment (ForestEckhardt) [reply text/0027-co-locate-all-rfcs.md:23]: Yes so like `Go`, `Java`, `Tooling`, etc.

**Review by fg-j — APPROVED** [review]

File: text/0027-co-locate-all-rfcs.md (hunk 2 above)
Comment (ForestEckhardt) [reply text/0027-co-locate-all-rfcs.md]: Added a link to the teams in Github.

**Comment by ForestEckhardt** [comment]
> @ForestEckhardt heads up there are 2 RFCs on this pull request. was that intentional?

@fg-j Thanks for pointing this out. It should be fixed now!

File: text/0027-co-locate-all-rfcs.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  easier commiting of RFCs by maintainers and contributors because they would
+  not need to create a fork of the repository to create an RFC. This still puts
+  a burden on individuals that would like to keep track of the going ons of
+  lots of teams.
+
+## Unresolved Questions and Bikeshedding
+
+- Is there a way to set up permissions to allow for project team members to
+  open up RFCs inside their respective language directories without having to
+  fork the RFCs repository? Should we just give all maintainers and
```
Comment (ekcasey) [inline text/0027-co-locate-all-rfcs.md:69]: Just curious, why is pushing a branch directly to the RFCs repo preferable to a fork?

**Review by ekcasey — APPROVED** [review]
Do we want to add labels to team-specific RFCs denoting the team for easy filtering. E.g
- team/go
- team/java
or something like that? This could be manually or potentially automated.

**Comment by fg-j** [comment]
I'd to see something added to the implementation section about how buildpacks' repos can point users to the relevant RFCs. @sophiewigmore you mentioned this at the Working Group meeting as well.

File: text/0027-co-locate-all-rfcs.md (hunk 4 above)
Comment (ForestEckhardt) [reply text/0027-co-locate-all-rfcs.md:69]: Not necessarily sure it is much easier, it is just the workflow that they are currently using. I think it is reasonable to fork just wanted to square away what the permissions are going to look like.

**Comment by ForestEckhardt** [comment]
> I'd to see something added to the implementation section about how buildpacks' repos can point users to the relevant RFCs. @sophiewigmore you mentioned this at the Working Group meeting as well.

@fg-j Hopefully the latest push squared away parts of this question. Let me know if there is anymore that you want me to add.

**Comment by ForestEckhardt** [comment]
Calling all maintainers! Please leave some form of approval of this RFC as a comment on this PR. I will try keeping which groups have reached quorum up-to-date by checking them off of this list.
* [ ] @paketo-buildpacks/nodejs-maintainers 
* [ ] @paketo-buildpacks/go-maintainers 
* [ ] @paketo-buildpacks/dotnet-core-maintainers 
* [ ] @paketo-buildpacks/php-maintainers 
* [ ] @paketo-buildpacks/java-buildpacks 
* [ ] @paketo-buildpacks/tooling-maintainers 
* [ ] @paketo-buildpacks/content-maintainers 
* [ ] @paketo-buildpacks/ruby-maintainers 
* [ ] @paketo-buildpacks/builders-maintainers
* [ ] @paketo-buildpacks/web-servers-maintainers 
* [ ] @paketo-buildpacks/stacks-maintainers 
* [ ] @paketo-buildpacks/dependencies-maintainers 
* [ ] @paketo-buildpacks/utilities-maintainers

**Comment by sophiewigmore** [comment]
I am good with this proposal.
I am a member of @paketo-buildpacks/tooling-maintainers / @paketo-buildpacks/dotnet-core-maintainers  / @paketo-buildpacks/ruby-maintainers

**Comment by martyspiewak** [comment]
I approve of this proposal!
I am a member of
@paketo-buildpacks/stacks-maintainers 
@paketo-buildpacks/dependencies-maintainers

**Comment by dmikusa-pivotal** [comment]
I approve of this proposal!
I am a member of:

 @paketo-buildpacks/php-maintainers
 @paketo-buildpacks/java-buildpacks

**Comment by fg-j** [comment]
I approve as a member of:

- @paketo-buildpacks/builders-maintainers
- @paketo-buildpacks/dotnet-core-maintainers and
- @paketo-buildpacks/go-maintainers

**Comment by dumez-k** [comment]
I approve of this proposal. 

I am a member of:
@paketo-buildpacks/stacks-maintainers
@paketo-buildpacks/builders-maintainers
@paketo-buildpacks/dependencies-maintainers

**Comment by arjun024** [comment]
Approve as a member of:
@paketo-buildpacks/nodejs-maintainers 
@paketo-buildpacks/php-maintainers
@paketo-buildpacks/utilities-maintainers 
@paketo-community/python-maintainers
@paketo-community/web-server-maintainers

**Comment by ForestEckhardt** [comment]
@paketo-buildpacks/steering-committee We have received a super majority of acceptance from all maintainers of existing projects.

**Review by ryanmoran — APPROVED** [review]

**Pull request merged**